packages/io/stl-serializer/index.js

```diff
@@ -1,59 +1,43 @@
 /*
-JSCAD Object to STL Format Serialization
+JSCAD Geometry to STL Format Serialization
 
 ## License
 
-Copyright (c) 2018 JSCAD Organization https://github.com/jscad
+Copyright (c) 2018-2019 JSCAD Organization https://github.com/jscad
 
 All code released under MIT license
 
 Notes:
-1) CAG conversion to:
+1) geom2 conversion to:
      none
-2) CSG conversion to:
+2) geom3 conversion to:
      STL mesh
-3) Path2D conversion to:
+3) path2 conversion to:
      none
 */
 
-const {serializeBinary} = require('./CSGToStlb')
-const {serializeText} = require('./CSGToStla')
+const { geometry, utils } = require('@jscad/modeling')
 
-const {ensureManifoldness} = require('@jscad/io-utils')
-const {toArray} = require('@jscad/io-utils/arrays')
-const {isCSG} = require('@jscad/csg')
+const { serializeBinary } = require('./CSGToStlb')
+const { serializeText } = require('./CSGToStla')
 
 const mimeType = 'application/sla'
 
-const serialize = (...params) => {
-  let options = {}
-  let objects
-  if (params.length === 0) {
-    throw new Error('no arguments supplied to serialize function !')
-  } else if (params.length === 1) {
-    // assumed to be object(s)
-    objects = Array.isArray(params[0]) ? params[0] : params
-  } else if (params.length > 1) {
-    options = params[0]
-    objects = params[1]
-  }
-  // make sure we always deal with arrays of objects as inputs
-  objects = toArray(objects)
-
+const serialize = (options, ...objects) => {
   const defaults = {
     binary: true,
     statusCallback: null
   }
   options = Object.assign({}, defaults, options)
 
-  // only use valid CSG objects
-  let csgs = []
-  objects.forEach(function (object, i) {
-    if (isCSG(object) & object.polygons.length > 0) {
-      csgs.push(ensureManifoldness(object))
-    }
-  })
-  return options.binary ? serializeBinary(csgs, options) : serializeText(csgs, options)
+  objects = utils.flatten(objects)
+
+  // convert only 3D geometries
+  objects = objects.filter((object) => geometry.geom3.isA(object))
+
+  if (objects.length === 0) throw new Error('the given objects cannot be converted')
```

Comment:
A bit more information would be helpfull in the message imho:
'No valid 3d geometry found to be serialized to stl data`